src/utils/applyWalletActions.js

```diff
@@ -0,0 +1,34 @@
+import Backend from './backend';
+import { createDeepLinkUrl } from './util';
+
+export const applyBackendChanges = async (method, request, requestParams, cb) => {
+  const args = {
+    sendProfileData: [...requestParams, request],
+    pinItem: [...requestParams, request],
+    unPinItem: [...requestParams, request],
+    sendPostReport: [request],
```

Comment:
This method shouldn't be aware of methods specific for components, otherwise, extraction is not so useful, because you have to modify this file to call an additional method in a new component